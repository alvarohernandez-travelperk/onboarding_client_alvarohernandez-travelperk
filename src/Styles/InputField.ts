import styled from 'styled-components'

export const Input = styled.input`
  border: 1px solid #eaeaea;
  border-radius: 5px;
  height: 40px;
  padding: 0px 8px;
  width: 200px;
  &:active, &:focus &:focus-visible{
    color: #0e1012;
    border-color: #0e1012;
  }
`;

export const InputLabel = styled.label`
  color: #0e1012;
  background-color: transparent;
  text-decoration: none;
  font-weight: 400;
  padding-right: 8px;
`;

export const InputBox = styled.span`
  margin: 10px 0px;
`;