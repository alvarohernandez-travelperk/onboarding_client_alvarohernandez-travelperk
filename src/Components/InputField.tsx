import React from 'react'
import { Input, InputLabel, InputBox } from '../Styles/InputField'


interface InputFieldProps {
  placeholder: string;
  title: string;
  value: string;
  handleInput: React.ChangeEventHandler<HTMLInputElement>;
}

export default function InputField({ placeholder, title, value, handleInput }: InputFieldProps) {
  return (
    <InputBox>
      <InputLabel>{title} :</InputLabel>
      <Input
        onChange={handleInput}
        type="text"
        value={value}
        placeholder={placeholder}></Input>
    </InputBox>
  )
}
