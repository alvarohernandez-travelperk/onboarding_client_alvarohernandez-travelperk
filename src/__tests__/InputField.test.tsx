import React from 'react';
import { render, cleanup } from '@testing-library/react';
import InputField from '../Components/InputField';

const handleInput = jest.fn();

afterEach(cleanup);

it("Snapshot Test Input Component", () => {
  const { asFragment } = render(<InputField
    title="title"
    placeholder="placeholder"
    value="value"
    handleInput={handleInput} />);
  expect(asFragment).toMatchSnapshot();
});

it("Renders right label text", () => {
  const { getByTestId, getByText  } = render(<InputField
    title="title-label"
    placeholder="placeholder"
    value="value"
    handleInput={handleInput} />);
  expect(getByTestId("input-label")).toHaveTextContent("title-label");
});


