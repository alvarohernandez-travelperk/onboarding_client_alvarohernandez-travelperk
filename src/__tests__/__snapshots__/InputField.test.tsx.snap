// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test Input Component 1`] = `[Function]`;
